### PR TITLE
Made spin gate names consistent

### LINKED ISSUE #21: Consistency of circuit instructions and gate names
There is some inconsistency in the naming of the circuit instructions, the string name of the gate and the gate name. For example:
```python
class LYGate(SpinGate):
    r"""Rotation of the collective spin of a cold atomic Bose-Einstein
    condensate around the y-axis.

    The generating Hamiltonian of the LY gate is

    :math:`H = \omega L_y`

    where :math:`\omega` is the free gate parameter

    **Circuit symbol:**

    .. parsed-literal::

             ┌────────────┐
        q_0: ┤ RLY(omega) ├
             └────────────┘
    """

    def __init__(self, omega, label=None):
        """Create new RLY gate."""
        super().__init__("rLy", 1, [omega], label=label)

    @property
    def generator(self) -> SpinOp:
        r"""The generating Hamiltonian of the LY gate."""
        return float(self.params[0]) * SpinOp("Y")


@add_gate
def ly(self, omega, wire):
    # pylint: disable=invalid-name
    """add the RLY gate to a QuantumCircuit"""
    return self.append(LYGate(omega), [wire], [])
```
should instead by
```python
class RLYGate(SpinGate):
    r"""Rotation of the collective spin of a cold atomic Bose-Einstein
    condensate around the y-axis.

    The generating Hamiltonian of the LY gate is

    :math:`H = \omega L_y`

    where :math:`\omega` is the free gate parameter

    **Circuit symbol:**

    .. parsed-literal::

             ┌────────────┐
        q_0: ┤ RLY(omega) ├
             └────────────┘
    """

    def __init__(self, omega, label=None):
        """Create new RLY gate."""
        super().__init__("rLy", 1, [omega], label=label)

    @property
    def generator(self) -> SpinOp:
        r"""The generating Hamiltonian of the RLYGate."""
        return float(self.params[0]) * SpinOp("Y")


@add_gate
def rly(self, omega, wire):
    # pylint: disable=invalid-name
    """add the RLY gate to a QuantumCircuit"""
    return self.append(RLYGate(omega), [wire], [])
```
This also means adapting the notebooks to this notation.

### PR DESCRIPTION
**Pull request by eggerdj**
The naming between the spin gates, the decorated circuit, and the spin gate display name has been made consistent. This fixes #21

### CODE REVIEW

**Review by LaurinFischer — APPROVED** [review]
Looks good to me, I'm completely on board with the new naming conventions.

**Comment by LaurinFischer** [comment]
Sorry, I tried to remove my above approval because I spotted some points after the initial review... 

- Regarding the discussion in #21, a point that Fred made is that the gate names (which eventually end up in the JSON via `circuit_to_cold_atom`) should be all lowercase like the circuit instructions, so `rlx` instead of `rLx`. I would agree with this. 

- Fred also suggested to remove the `r` from all two-spin gates, in order not to clutter the notation too much. I'd also agree with this. 

Once we agree and this is implement we can merge this and close #21.

**Comment by LaurinFischer** [comment]
I found one more place where the names should be changed which is the configuration of the `collective_spin_simulator` backend. I updated that in the previous commit.

**Review by LaurinFischer — COMMENTED** [review]
I think the new gate names are now in place everywhere. Let's wait for Fred's approval of the new conventions before merging.

File: qiskit_cold_atom/providers/collective_spin_backend.py
Comment (fretchen) [inline qiskit_cold_atom/providers/collective_spin_backend.py:48]: perfect.

**Comment by fretchen** [comment]
LGTM.

**Review by LaurinFischer — APPROVED** [review]

**Pull request merged**
